lib/page-object/loads_platform.rb

```diff
@@ -15,12 +15,31 @@ module LoadsPlatform
     # @returns [PageObject] 
     #
     def load_platform(browser, adapters)
-      adapters.each_value { |adapter|
-        return adapter.create_page_object(browser) if adapter.is_for?(browser)
+      adapter_for(browser,adapters).create_page_object(browser)
+    end
+
+    def browser_for root,adapters
```

Comment:
since we are now identifying root elements and browsers, added those concerns to the platform loading code. First we find the required platform, then we ask for the element or browser.